native/src/main/java/io/ballerina/observe/metrics/newrelic/NewRelicMetricsReporter.java

```diff
@@ -0,0 +1,189 @@
+import io.ballerina.runtime.observability.metrics.Gauge;
+import io.ballerina.runtime.observability.metrics.Metric;
+import io.ballerina.runtime.observability.metrics.MetricConstants;
+import io.ballerina.runtime.observability.metrics.MetricId;
+import io.ballerina.runtime.observability.metrics.PercentileValue;
+import io.ballerina.runtime.observability.metrics.PolledGauge;
+import io.ballerina.runtime.observability.metrics.Snapshot;
+import io.ballerina.runtime.observability.metrics.Tag;
+
+import java.io.PrintStream;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static io.ballerina.observe.metrics.newrelic.ObserveNativeImplConstants.EXPIRY_TAG;
+import static io.ballerina.observe.metrics.newrelic.ObserveNativeImplConstants.PERCENTILE_TAG;
+
+/**
+ * This is the New Relic metric reporter class.
+ */
+public class NewRelicMetricsReporter {
+    private static final PrintStream console = System.out;
+    private static final String METRIC_REPORTER_ENDPOINT = "https://metric-api.newrelic.com/metric/v1";
+    private static final int SCHEDULE_EXECUTOR_INITIAL_DELAY = 0;
+
+    public static void sendMetrics(BString apiKey, int metricReporterFlushInterval,
+                                         int metricReporterClientTimeout) {
+        // create a TelemetryClient with an HTTP connect timeout of 10 seconds.
+        TelemetryClient telemetryClient =
+                TelemetryClient.create(
+                        () -> new OkHttpPoster(Duration.of(metricReporterClientTimeout, ChronoUnit.MILLIS)),
+                        apiKey.getValue());
+        Attributes commonAttributes = null;
+        try {
+            commonAttributes = new Attributes()
+                    .put("host", InetAddress.getLocalHost().getHostName())
+                    .put("language", "ballerina");
+        } catch (UnknownHostException e) {
+            console.println("Error while getting the host name");
+        }
+
+        // Create a ScheduledExecutorService with a single thread
+        ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+
+        // Schedule a task to run every 1 second with an initial delay of 0 seconds
+        Attributes finalCommonAttributes = commonAttributes;
+        executorService.scheduleAtFixedRate(() -> {
+            MetricBuffer metricBuffer = generateMetricBuffer(finalCommonAttributes);
+            MetricBatch batch = metricBuffer.createBatch();
+            telemetryClient.sendBatch(batch);
+        }, SCHEDULE_EXECUTOR_INITIAL_DELAY, metricReporterFlushInterval, TimeUnit.MILLISECONDS);
+
+        console.println("ballerina: started publishing metrics to New Relic on " + METRIC_REPORTER_ENDPOINT);
```

Comment:
Is it possible to pass to the ballerina level and log the message instead of using console print?

Reply:
For traces, it is not possible.